File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        if n == 1: return 1
        sums, start, end = n * (n + 1) / 2, 0, n
        while start < end: 
            mid = start + (end - start) // 2
            if mid * mid == sums: return mid
            elif mid * mid > sums: end = mid - 1
            else: start = mid + 1
        return -1

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.java
class Solution {
    public int pivotInteger(int n) {
        if(n == 1) return 1;
        int sum = n * (n + 1) / 2, start = 0, end = n;
        while(start < end) {
            int mid = start + (end - start) / 2;
            if(mid * mid == sum) return mid;
            else if(mid * mid > sum) end = mid - 1;
            else start = mid + 1;
        }
        return -1;
    }
}